sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusProcessorClient.cs

```diff
@@ -704,22 +704,23 @@ public virtual async Task StopProcessingAsync(CancellationToken cancellationToke
             ProcessingOptions options,
             CancellationToken cancellationToken)
         {
+            IList<Task> tasks = new List<Task>();
             while (!cancellationToken.IsCancellationRequested)
             {
                 await MessageHandlerSemaphore.WaitAsync(cancellationToken).ConfigureAwait(false);
 
                 try
                 {
-                    Task _ = GetAndProcessMessagesAsync(options, cancellationToken);
+                    tasks.Add(GetAndProcessMessagesAsync(options, cancellationToken));
```

Comment:
revert

Reply:
actually nevermind. This somehow didn't make it in from previous PR. It should be there.